### PR TITLE
feat: types

### PR DESCRIPTION
**Pull request by hugomrdias**
Adds types support

### CODE REVIEW

**Comment by hugomrdias** [comment]
ready

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - uses: actions/checkout@v2
+      - run: npm install
+      - run: npx aegir test -t browser -t webworker --bail
+  test-firefox:
+    needs: check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: npm install
+      - run: npx aegir test -t browser -t webworker --bail -- --browsers FirefoxHeadless
```
Comment (achingbrain) [inline .github/workflows/main.yml:49]:
```suggestion
```

Not sure this needs to run tests in the browser?

**Review by achingbrain — APPROVED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - uses: gozala/typescript-error-reporter-action@v1.0.8
+    - run: npx aegir build --no-bundle
+    - run: npx aegir dep-check
+  test-node:
+    needs: check
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        node: [12, 14]
```
Comment (achingbrain) [inline .github/workflows/main.yml:26]:
```suggestion
        node: [14, 15]
```

This should be 14 & 15 as that's LTS/Current.  Going through and updating all of these repos every time there's a new node version is going to be an enormous pain, can it be automated?

Comment (hugomrdias) [reply .github/workflows/main.yml:26]: https://github.com/actions/setup-node/issues/26

Comment (achingbrain) [reply .github/workflows/main.yml:26]: That issue looks like it's stalled and the [linked PR](https://github.com/actions/setup-node/pull/58) is stale and also contains a bunch of irrelevant changes which will harm its chances of being merged.

I think we'll need to do something ourselves for the time being.

**Pull request merged**